Source/Widgets/Cesium3DTilesInspector/Cesium3DTilesInspectorViewModel.js

```diff
@@ -0,0 +1,880 @@
+        this.pickActive = !this.pickActive;
+    };
+
+    /**
+     * Toggles the inspector visibility
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleInspector = function() {
+        this.inspectorVisible = !this.inspectorVisible;
+    };
+
+    /**
+     * Toggles the visibility of the tileset section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleTileset = function() {
+        this.tilesetVisible = !this.tilesetVisible;
+    };
+
+    /**
+     * Toggles the visibility of the display section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleDisplay = function() {
+        this.displayVisible = !this.displayVisible;
+    };
+
+    /**
+     * Toggles the visibility of the update section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleUpdate = function() {
+        this.updateVisible = !this.updateVisible;
+    };
+
+    /**
+     * Toggles the visibility of the logging section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleLogging = function() {
+        this.loggingVisible = !this.loggingVisible;
+    };
+
+    /**
+     * Toggles the visibility of the style section
+     */
+    Cesium3DTilesInspectorViewModel.prototype.toggleStyle = function() {
+        this.styleVisible = !this.styleVisible;
+    };
+
+    /**
+     * Trims tile cache
+     */
+    Cesium3DTilesInspectorViewModel.prototype.trimTilesCache = function() {
+        if (defined(this._tileset)) {
+            this._tileset.trimLoadedTiles();
+        }
+    };
+
+    /**
+     * Compiles the style in the style editor
+     */
+    Cesium3DTilesInspectorViewModel.prototype.compileStyle = function() {
+        if (defined(this._style)) {
+            if (this.styleString !== JSON.stringify(this._style.style)) {
+                if (defined(this._tileset)) {
+                    var old = this._tileset.style;
+                    this._editorError = '';
+                    try {
+                        if (this.styleString.length === 0) {
+                            this.styleString = '{}';
+                        }
+                        var style = new Cesium3DTileStyle(JSON.parse(this.styleString));
+                        this._tileset.style = style;
+                        this._style = style;
+                        this._tileset.update(this._scene.frameState);
+                    } catch (err) {
+                        this._tileset.style = old;
+                        this._style = old;
+                        this._editorError = err.toString();
+                    }
+
+                    // set feature again so pick coloring is set
+                    this.feature = this._feature;
+                }
+            }
+        }
+    };
+
+    /**
+     * Handles key press events on the style editor
+     */
+    Cesium3DTilesInspectorViewModel.prototype.styleEditorKeyPress = function(sender, event) {
+        if (event.keyCode === 9) { //tab
```

Comment:
This block is crying for a real editor, like CodeMirror, but I guess there's nothing stopping someone from putting CodeMirror on top of this view model and ignoring this function.